java/daikon/dcomp/DCInstrument.java

```diff
@@ -1928,26 +2015,87 @@ InstructionList handle_invoke(InvokeInstruction invoke) {
         }
       }
 
+      // if we are not using the instrumented JDK, then we need to track down the
+      // actual target of an INVOKEVIRTUAL to see if it has been instrumented or not.
       if (invoke instanceof INVOKEVIRTUAL) {
-        if (!jdk_instrumented) {
+        if (!jdk_instrumented
+            && !mgen.getName().equals("equals_dcomp_instrumented")
+            && !is_object_clone(method_name, ret_type, arg_types)
+            && !is_object_equals(method_name, ret_type, arg_types)) {
+
           // debug code
-          // System.out.printf("invoke virtual: %s : %s%n", classname, method_name);
-          // System.out.printf("super class: %s%n", gen.getSuperclassName());
-          // Technically, we should verify the target class has super class
-          // of java.lang.Enum. But that can be difficult if we haven't already
-          // processed that class. And since the worst that happens is we
-          // loose some tag interactions, we just go ahead.
-          if (method_name.equals("ordinal")) {
-            callee_instrumented = false;
-          }
+          // System.out.println("method: " + method_name);
+          // System.out.println("arg_types: " + Arrays.toString(arg_types));
+          // System.out.printf("invoke host: %s%n", gen.getClassName() + "." + mgen.getName());
+
+          @ClassGetName String targetClassname = classname;
+          mainloop:
+          while (true) {
```

Comment:
It would be helpful to document the purpose of this `while` loop.  Is its goal to set a particular variable, such as `callee_instrumented` and/or `classname`?